packages/eslint-plugin-office-addins/test/rules/load-object-before-read.test.ts

```diff
@@ -0,0 +1,136 @@
+import { ESLintUtils } from '@typescript-eslint/experimental-utils'
+import rule from '../../src/rules/load-object-before-read';
+
+const ruleTester = new ESLintUtils.RuleTester({
+  parser: '@typescript-eslint/parser',
+});
+
+ruleTester.run('load-object-before-read', rule, {
+  valid: [ 
```

Comment:
Is it possible and valid to have:

  var myRange = context.workbook.worksheets.getItem(sheetName).getRange(rangeAddress);          
  myRange.load('address');
  myRange.address.someOtheProperty

or does the sync and load have to happen before the nested property can be accessed?

Reply:
The sync has to be called beforehand. 
This case should be handled by another rule: "Always call load or an object (followed by a sync) before reading it or one of its properties."

They almost detect the same thing, but the messages are different. I think the correct approach would be to merge both rules and show different messages?

Reply:
I was focused on the nested property case.  I think the sync would be a separate rule.

Reply:
@millerds after doing some research, I found that there are no navigation properties (which would lead to nested properties) coming from the get* functions.
So we don't need to do these checks, because there will never be nested properties.

Reply:
Asked for clarification on this rule and there is a slash that needs to be used for those. I added test cases for that and the added the necessary code.